tests/Test/Xrefcheck/AnchorsInHeadersSpec.hs

```diff
@@ -0,0 +1,31 @@
+{- SPDX-FileCopyrightText: 2019 Serokell <https://serokell.io>
+ -
+ - SPDX-License-Identifier: MPL-2.0
+ -}
+
+module Test.Xrefcheck.AnchorsInHeadersSpec where
+
+import qualified Data.ByteString.Lazy as BSL
+import Test.Hspec (Spec, describe, it, shouldBe)
+
+import Xrefcheck.Core
+import Xrefcheck.Scanners.Markdown
+
+spec :: Spec
+spec = do
+    describe "\"ignore link\" mode" $ do
+        it "Check \"ignore link\" performance" $ do
+            fi <- getFI "tests/markdowns/without-annotations/anchors_in_headers.md"
+            getAnchors fi `shouldBe` ["-some-stuff", "stuff-section"]
```

Comment:
```hs
"-some-stuff"
```

Where did the `-` come from? :thinking: 

I think we may have accidentally discovered another bug, because this is not considered valid (even though it should be):

```
## <a name="stuff-section"></a> Some stuff

[My link](#some-stuff)
```

```
=== Invalid references found ===

  ➥  In file tests/markdowns/without-annotations/anchors_in_headers.md
     bad reference (local) at src:9:1-22:
       - text: "My link"
       - link: 
       - anchor: some-stuff
     
     ⛀  Anchor 'some-stuff' is not present,
        did you mean -some-stuff (header II) at src:7:1-42?
```

Conversely, this is now valid (even though it shouldn't be):

```
## <a name="stuff-section"></a> Some stuff

[My link](#-some-stuff)
```

And if we remove the anchor tag, then the `-` prefix disappears and this is considered valid, as expected:

```
## Some stuff

[My link](#some-stuff)
```

Can you please open an issue to address this please?

Reply:
> Where did the - come from? 🤔

It comes from `headerToAnchor` which was added to close [#55](https://github.com/serokell/xrefcheck/issues/55). In this function, all spaces in heading are replaced with "-" to produce an anchor.

AFAIU when `## <a name="stuff-section"></a> Some stuff` is being parsed, " Some stuff" is considered as a heading instead of "Some stuff". When it's only the heading without HTML code, everything is okay. I tried to remove the space `## <a name="stuff-section"></a>Some stuff` and "some-stuff" became a valid anchor.

The first probable fix that comes to my mind is stripping the text in `nodeExtractText `:
```hs
nodeExtractText :: Node -> Text
nodeExtractText = T.strip . mconcat . map extractText . nodeFlatten
...
```
But perhaps it's a hack and there could be another solution.

Reply:
Ah that makes sense!

Yeah, I think, the `strip` solution makes sense, it seems to be what github does.

Reply:
Cool!

It just seems strange to me that `CMark` doesn't perform it by default.